### PR TITLE
NIFI-7917 Upgrade avro to 1.10 in nifi-nar-bundles/nifi-standard-serv…

### PR DESCRIPTION
**Pull request by adenes**
…ices/nifi-record-serialization-services-bundle/nifi-record-serialization-services

- Updated avro version to 1.10
- Replaced org.codehaus.jackson.* with com.fasterxml.jackson classes as previously the former was used and provided transitively through avro 1.8
- Added org.tukaani:xz:1.5 and org.xerial.snappy:snappy-java:1.1.1.3 dependencies explicitly as those were also coming through avro

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [x] Does your PR title start with **NIFI-XXXX** where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [x] Has your PR been rebased against the latest commit within the target branch (typically `main`)?

- [x] Is your initial contribution a single, squashed commit? _Additional commits in response to PR reviewer feedback should be made on this branch and pushed to allow change tracking. Do not `squash` or use `--force` when pushing to allow for clean monitoring of changes._

### For code changes:
- [ ] Have you ensured that the full suite of tests is executed via `mvn -Pcontrib-check clean install` at the root `nifi` folder?
- [ ] Have you written or updated unit tests to verify your changes?
- [x] Have you verified that the full build is successful on JDK 8?
- [ ] Have you verified that the full build is successful on JDK 11?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [ ] If applicable, have you updated the `LICENSE` file, including the main `LICENSE` file under `nifi-assembly`?
- [ ] If applicable, have you updated the `NOTICE` file, including the main `NOTICE` file found under `nifi-assembly`?
- [ ] If adding new Properties, have you added `.displayName` in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: nifi-nar-bundles/nifi-standard-services/nifi-record-serialization-services-bundle/nifi-record-serialization-services/pom.xml [hunk 1]
```diff
@@ -124,6 +124,16 @@
             <artifactId>guava</artifactId>
             <version>27.0.1-jre</version>
         </dependency>
+        <dependency>
+            <groupId>org.tukaani</groupId>
+            <artifactId>xz</artifactId>
```
Comment (simonbence) [inline nifi-nar-bundles/nifi-standard-services/nifi-record-serialization-services-bundle/nifi-record-serialization-services/pom.xml:129]: I think these two should be added to the NOTICE of the module as components.

**Comment by mtien-apache** [comment]
I noticed there are other modules that use Avro. Can these be upgraded as well?

- nifi-avro-record-utils
- nifi-hadoop-libraries-nar
- nifi-database-utils
- nifi-cassandra-processors
- nifi-hwx-schema-registry-service
- nifi-parquet-processors

**Comment by mattyb149** [comment]
The build fails due to some remaining references to `org.codehaus.jackson.map.ObjectMapper` which should be replaced with `com.fasterxml.jackson.databind.ObjectMapper`:

AtlasAPIV2ServerEmulator.java
ITReportLineageToAtlas.java
QueryCassandraTest.java
GetHDFSEvents.java
TestAvroRecordConverter.java
ConvertJSONToSQL.java
TestConvertJSONToSQL.java

**Comment by mattyb149** [comment]
Also two files refer to `org.codehaus.jackson.JsonNode` and should be changed to `com.fasterxml.jackson.databind.JsonNode`:

ConvertJSONToSQL
TestConvertJSONToSQL

**Comment by mattyb149** [comment]
`AtlasAPIV2ServerEmulator` refers to `DeserializationConfig.Feature` which needs to be replaced with `DeserializationFeature`

**Comment by mattyb149** [comment]
I took the liberty of making the changes and putting up a [commit](https://github.com/mattyb149/nifi/commit/5b58e082f632db5adbb67feca678b414edd2550f), feel free to cherry-pick if you wish.

**Pull request closed without merging**